### PR TITLE
Fix return of error adding artifacts to cache when images are built remotely

### PR DESCRIPTION
**Pull request by ricardo-larosa**
Fixes: https://github.com/GoogleContainerTools/skaffold/issues/4848

**Description**
Fix return error adding artifacts to cache when builds are not local

### CODE REVIEW

File: pkg/skaffold/build/cache/retrieve.go [hunk 1]
```diff
@@ -147,8 +147,11 @@ func maintainArtifactOrder(built []build.Artifact, artifacts []*latest.Artifact)
 }
 
 func (c *cache) addArtifacts(ctx context.Context, bRes []build.Artifact, hashByName map[string]string) error {
```
Comment (gsquared94) [inline pkg/skaffold/build/cache/retrieve.go:149]: you can simplify it a bit
```go
func (c *cache) addArtifacts(ctx context.Context, bRes []build.Artifact, hashByName map[string]string) error {
	for _, a := range bRes {
		entry := ImageDetails{}
		if c.imagesAreLocal {
			imageID, err := c.client.ImageID(ctx, a.Tag)
			if err != nil {
				return err
			}

			if imageID != "" {
				entry.ID = imageID
			}
		} else {
			ref, err := docker.ParseReference(a.Tag)
			if err != nil {
				return fmt.Errorf("parsing reference %q: %w", a.Tag, err)
			}
			entry.Digest = ref.Digest
		}
		c.cacheMutex.Lock()
		c.artifactCache[hashByName[a.ImageName]] = entry
		c.cacheMutex.Unlock()
	}
	return nil
}
```

Comment (ricardo-larosa) [reply pkg/skaffold/build/cache/retrieve.go:149]: ok

Comment (ricardo-larosa) [reply pkg/skaffold/build/cache/retrieve.go:149]: Thanks @gsquared94

**Review by tejal29 — APPROVED** [review]
Thanks for catching this  @ricardo-larosa 
The change looks good to me

**Pull request merged**